tcconfig/shaper/htb.py

```diff
@@ -112,10 +132,25 @@ def _add_rate(self):
         ]
 
         if bandwidth != upper_limit_rate:
+            # This sets the burst/cburst values to the default tc tries to set,
+            # but works around a bug in tc where values are rounded incorrectly.
+            
+            # Note that "tc class show" will still show burst that are too small,
+            # due to the inverse bug, but if you use "tc -r class show" you will see
+            # the actual size of the cbuffer in bytes.
+            mtu = 1600
+            rate = bandwidth.byte_per_sec
+            desired_burst = int(rate / get_hz() + mtu)
+            burst = desired_burst
+            while True:
+                if calc_xmitsize_true(rate, calc_xmittime_bug(rate, burst)) >= desired_burst:
+                    break
```

Comment:
Can you be sure that this calculation will not go into an infinite loop for any value?
Wouldn't it be safer to set a maximum number of loops?

Reply:
I've switched it to use `bisect_left` (following the inspiration of https://fanchenbao.medium.com/full-powered-binary-search-with-bisect-in-python-3-10-fb4a76110746) which is O(log(n)) and guaranteed to halt.